spyder_line_profiler/spyder/widgets.py

```diff
@@ -257,17 +349,26 @@ def analyze(self, filename=None, wdir=None, args=None, pythonpath=None,
 
     def select_file(self):
         self.redirect_stdio.emit(False)
+        try:    
+            curr_script = self.editor.get_current_filename()
+        except:
+            curr_script = None
+            
+        pwd = curr_script if curr_script else getcwd_or_home()
```

Comment:
`self.editor` is not defined so I guess this always goes to the expect path. If that is the case, this could be changed to:
```suggestion
        pwd = getcwd_or_home()
```